plugins/versionpress/src/Database/schema-readme.md

```diff
@@ -1,124 +1,244 @@
+
+```
+option:
+  value-references:
+    option_name@option_value:
+      theme_mods_*["nav_menu_locations"][/.*/]: term
+      theme_mods_*["header_image_data"]["attachment_id"]: post
+      theme_mods_*["custom_logo"]: post
+```
+
+
+### M:N references
+
+Some entities are in an M:N relationship like posts and term_taxonomies. The format to capture this is:
+
+```
+mn-references:
+  <junction_table_name>.<column_name>: <foreign_entity_name>
+```
+
+This is a concrete example from the post entity:
+
+```
+post:
+  mn-references:
+    term_relationships.term_taxonomy_id: term_taxonomy
+```
+
+One entity is considered "master" which is kind of arbitrary (technically, they are equal) but here, we decided that posts will store tags and categories, not the other way around. INI files of posts will store the references.
+
+References can also be prefixed with a tilde (`~`) which makes them virtual:
+
+```
+mn-references:
+  ~<junction_table_name>.<column_name>: <foreign_entity_name>
+```
+
+Virtual references are not stored in INI files but the relationships are checked during reverts. For example, when a revert would delete a category (revert of `term_taxonomy/create`) and there is some post referencing it, the operation would fail.
+
+
+### Parent references
+
+Some entities are stored within other entities, for example, postmeta are stored in the same INI file as their parent post. This is captured using a `parent-reference` property:
+
+```
+postmeta:
+  parent-reference: post_id
+  references:
+    post_id: post
+
+```
+
+This references one of the basic reference column names, not the final entity. The notation above reads "postmeta stores a parent reference in the `post_id` column and that points to the `post` entity".
+
 
 ## Frequently written entities
 
-Some entities are changed very often (view counter, akismet spam count, etc.). It is possible to save them once in a while.
-They are specified in section `frequently-written`. It's a list of selectors or combination of selector and custom interval. Default interval is `1hour`.
+Some entities are changed very often, e.g., view counters, Akismet spam count, etc. VersionPress only saves them once in a while and the `frequently-written` section influences this:
 
-    frequently-written:
-        - 'column_name: value'
-        - query: 'column1_name: value1 column2_name: value2'
-          interval: 5min
+```
+entity:
+  frequently-written:
+    - 'column_name: value'
+    - query: 'column1_name: value1 column2_name: value2'
+      interval: 5min
+```
 
-The interval is parsed by PHP function `strtotime`, so it can be whatever the function takes.
+The values in the `frequently-written` array can either be strings which are then interpreted as queries, or objects with `query` and `interval` keys. The interval is parsed by the `strtotime()` PHP function and the default value is one hour.
```

Comment:
More or less. They don't support range search for date and each key (DB column) can appear at most once.